src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -237,6 +222,27 @@ public QueryExecutor(DbExceptionParser dbExceptionParser,
             return result;
         }
 
+        /// <summary>
+        /// Creates and setups a TConnection to the data source of given name
+        /// </summary>
+        /// <param name="dataSourceName">The data source name</param>
+        /// <returns>A connection to the data source</returns>
+        /// <exception cref="DataApiBuilderException">Exeption thrown if the data source could not be found</exception>
+        public virtual TConnection CreateConnection(string dataSourceName)
+        {
+            if (!ConnectionStringBuilders.ContainsKey(dataSourceName))
+            {
+                throw new DataApiBuilderException("Query execution failed. Could not find datasource to execute query against", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
```

Comment:
nit: keep 100 char limit, split the line in multiple and honor the indentation

Reply:
Done!